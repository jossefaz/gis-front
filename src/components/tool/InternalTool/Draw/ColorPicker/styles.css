.color {
    width: 36px;
    height: 14px;
    border-radius: 2px;
}
.swatch {
    padding: 5px;
    background: #fff;
    border-radius: 1px;
    box-shadow: 0 0 0 1px rgba(0,0,0,.1);
    display: inline-block;
    cursor: pointer
}
.popover {
    position: absolute;
    z-index: 2
}
.cover {
    position: fixed;
    top: 0px;
    right: 0px;
    bottom: 0px;
    left: 0px
}

.numberInput .ui.button {
    background-color: #2185d0;
    border-radius: 0 !important;
    color: white;
  }
  
  .numberInput .ui.input > input {
    border-color: #2185d0;
    color: #2185d0;
    font-weight: bold;
    width: 50px;
  }

  .margin0 {
      margin: 0 !important;
      display: flex;
  }


  .labels  {
    font-size: .85em;
    padding-top: .2em;
    padding-left: 15px;
    font-weight: bold;
  }