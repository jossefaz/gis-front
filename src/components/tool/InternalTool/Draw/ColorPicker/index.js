import React from 'react'
import { SketchPicker } from 'react-color'
import './styles.css'

class SketchExample extends React.Component {
    state = {
        displayColorPicker: false,
        color: {
            r: '241',
            g: '112',
            b: '19',
            a: '1',
        },
    };

    componentDidMount() {
        if (this.props.defaultColor)
            this.setState({ color: this.props.defaultColor })
    }

    handleClick = () => {
        this.setState({ displayColorPicker: !this.state.displayColorPicker })
    };

    handleClose = () => {
        this.setState({ displayColorPicker: false })
    };

    handleChange = (color) => {
        this.setState({ color: color.rgb })
        this.props.onColorChange(color.rgb)
    };

    render() {

        return (
            <div>
                <div className="swatch" onClick={this.handleClick}>
                    <div className="color" style={{ background: `rgba(${this.state.color.r}, ${this.state.color.g}, ${this.state.color.b}, ${this.state.color.a})` }} />
                </div>
                {this.state.displayColorPicker ? <div className="popover">
                    <div className="cover" onClick={this.handleClose} />
                    <SketchPicker color={this.state.color} onChange={this.handleChange} />
                </div> : null}

            </div>
        )
    }
}

export default SketchExample