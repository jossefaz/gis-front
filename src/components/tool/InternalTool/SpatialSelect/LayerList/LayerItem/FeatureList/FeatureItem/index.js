import React, { useState } from "react";
import { Checkbox } from "semantic-ui-react";
import VectorLayerRegistry from "../../../../../../../../utils/vectorlayers";
import IconButton from "../../../../../../../UI/Buttons/IconButton";
import { Confirm } from "semantic-ui-react";
import styles from "../../../../../../../../nessMapping/mapStyle";
import { getFeatureProperties } from "../../../../../../../../nessMapping/api";

const FeatureItem = ({ feature, style, removeFeature }) => {
  const [Checked, setChecked] = useState(true);
  const [Modal, setModal] = useState(false);
  const erasefeature = {
    content: "? האם באמת למחוק את היישות",
    confirmBtn: "כן",
    cancelBtn: "לא",
  };
  const toggleFeature = () => {
    if (Checked) {
      feature.setStyle(styles.HIDDEN);
    } else {
      feature.setStyle(style);
    }
    setChecked(!Checked);
  };

  const removeLocalFeature = () => {
    feature.setStyle(styles.HIDDEN);
    removeFeature(feature);
    setModal(false);
  };

  return (
    <React.Fragment>
      <div className="displayFlex">
        <Checkbox
          label={feature.getId()}
          onChange={toggleFeature}
          checked={Checked}
        />
        <IconButton
          className={`ui icon button pointer negative`}
          onClick={() => setModal(true)}
          icon="trash-alt"
          size="xs"
        />
      </div>

      <Confirm
        open={Modal}
        size="mini"
        content={erasefeature.content}
        cancelButton={erasefeature.cancelBtn}
        confirmButton={erasefeature.confirmBtn}
        onCancel={() => setModal(false)}
        onConfirm={removeLocalFeature}
      />
    </React.Fragment>
  );
};

export default FeatureItem;
