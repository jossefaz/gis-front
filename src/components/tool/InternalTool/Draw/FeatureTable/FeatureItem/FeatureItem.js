import React, { useState } from "react";
import ColorPicker from "../../../../../UI/ColorPicker/ColorPicker";
import { generateNewPolygonStyle } from "../../../../../../utils/func";
import { printCoreState } from "../../../../../../core/monitor";
import IconButton from "../../../../../UI/Buttons/IconButton";
import API from "../../../../../../core/api";
import { Table, Form, Input } from "semantic-ui-react";
import { deleteGeometry } from "../../../../../../services/persistentGeometry/api";
import "./style.css";
const { unhighlightFeature } = API.features;
export default (props) => {
  const [fillColor, setFillColor] = useState({
    r: "154",
    g: "111",
    b: "222",
    a: "0.2",
  });
  const [outlineWidth, setOutlineWidth] = useState(2);

  const editStyle = {
    stroke: `rgba(19,187,254,.87)`,
    fill: `rgba(255,255,255,0.7)`,
  };

  const [strokeColor, setStrokeColor] = useState(props.defaultColor);
  const removeFeature = () => {
    const feature = getFeature();
    // deleteGeometry(feature);
    props.source.removeFeature(feature);
    unhighlightFeature();
    props.deleteLastFeature(props.fid);
  };
  const edit = () => {
    props.onOpenEditSession(props.fid);
  };
  const getFeature = () => props.source.getFeatureById(props.fid);
  const updateStyle = () => {
    if (getFeature()) {
      const currentStroke = `rgba(${strokeColor.r},${strokeColor.g},${strokeColor.b},${strokeColor.a})`;
      const currentFill = `rgba(${fillColor.r},${fillColor.g},${fillColor.b},${fillColor.a})`;
      getFeature().setStyle(
        generateNewPolygonStyle(currentFill, currentStroke, outlineWidth)
      );
    }
  };
  if (!props.editSession.status) {
    updateStyle();
  } else {
    if (props.editSession.current == props.fid) {
      getFeature().setStyle(
        generateNewPolygonStyle(editStyle.fill, editStyle.stroke, 3)
      );
    } else {
      updateStyle();
    }
  }
  printCoreState();
  return (
    <React.Fragment>
      <Table.Cell>
        <p>{props.index + 1}</p>
      </Table.Cell>
      <Table.Cell>
        <div className="displayFlex">
          <label className="labels">קו חיצוני :</label>
          <ColorPicker
            withWidth
            initialWidth={outlineWidth}
            onWidthChange={(newWidth) => setOutlineWidth(newWidth)}
            onColorChange={(newStrokeColor) => setStrokeColor(newStrokeColor)}
            defaultColor={strokeColor}
          />
        </div>
        <div className="displayFlex">
          <label className="labels">מילוי :</label>
          <ColorPicker
            onColorChange={(newFillColor) => setFillColor(newFillColor)}
            defaultColor={fillColor}
          />
        </div>
      </Table.Cell>
      <Table.Cell>
        <IconButton
          className="ui icon button pointer negative"
          onClick={removeFeature}
          icon="trash-alt"
          size="lg"
        />
        <IconButton
          className="ui icon button pointer positive"
          onClick={edit}
          icon="edit"
          size="lg"
        />
      </Table.Cell>
    </React.Fragment>
  );
};
