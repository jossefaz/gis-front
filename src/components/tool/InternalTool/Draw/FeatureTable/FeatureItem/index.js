import React from 'react'
import ColorPicker from '../../ColorPicker'
import { generateNewStyle } from '../../../MeasureDistance/func'
export default (props) => {

    const onColorChange = ({ r, g, b, a }) => {
        const feature = props.source.getFeatureById(props.fid)
        feature.setStyle(generateNewStyle(`rgba(${r},${g},${b},${a})`))

    }
    const FeatureItem = () => {
        return props.properties ? props.properties.map(pr => {
            return <p key={pr}>{pr}</p>
        }) : <p key={props.fid}>{props.fid}</p>
    }
    return (
        <React.Fragment>
            <FeatureItem />
            <ColorPicker onColorChange={onColorChange} defaultColor={props.defaultColor} />
        </React.Fragment>

    )
}
