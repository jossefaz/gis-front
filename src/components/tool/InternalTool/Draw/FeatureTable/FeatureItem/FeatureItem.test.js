import { shallow, mount } from "enzyme";
import React, { useState as useStateMock } from "react";
import FeatureItem from "./FeatureItem";
import ColorPicker from "../../ColorPicker/ColorPicker";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));
describe("FeatureItem component", () => {
  const mockgetFeatureById = jest.fn((x) => 0);

  const setState = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setState]);
  });

  it("expect to match snapshot", () => {
    const FeatureItemSh = shallow(
      <FeatureItem
        editSession={{ status: false }}
        source={{ getFeatureById: mockgetFeatureById }}
      />
    );
    expect(FeatureItemSh).toMatchSnapshot();
  });

  //   it("function OutlineWidth to be updated", () => {
  //     const FeatureItemSh2 = mount(
  //       <FeatureItem
  //         editSession={{ status: false }}
  //         source={{ getFeatureById: mockgetFeatureById }}
  //       />
  //     );
  //     expect(setState).toHaveBeenCalledTimes(1);
  //   });
});
