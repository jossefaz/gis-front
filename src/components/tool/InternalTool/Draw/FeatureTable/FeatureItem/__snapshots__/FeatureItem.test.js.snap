// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureItem component expect to match snapshot 1`] = `
<Fragment>
  <TableCell
    as="td"
  >
    <p>
      NaN
    </p>
  </TableCell>
  <TableCell
    as="td"
  >
    <div
      className="displayFlex"
    >
      <label
        className="labels"
      >
        קו חיצוני :
      </label>
      <SketchExample
        initialWidth={2}
        onColorChange={[Function]}
        onWidthChange={[Function]}
        withWidth={true}
      />
    </div>
    <div
      className="displayFlex"
    >
      <label
        className="labels"
      >
        מילוי :
      </label>
      <SketchExample
        defaultColor={
          Object {
            "a": "0.2",
            "b": "222",
            "g": "111",
            "r": "154",
          }
        }
        onColorChange={[Function]}
      />
    </div>
  </TableCell>
  <TableCell
    as="td"
  >
    <_default
      className="ui icon button pointer negative"
      icon="trash-alt"
      onClick={[Function]}
      size="lg"
    />
    <_default
      className="ui icon button pointer positive"
      icon="edit"
      onClick={[Function]}
      size="lg"
    />
  </TableCell>
</Fragment>
`;
